pkg/wallet/wallet.go

```diff
@@ -63,12 +68,71 @@ func (w *BaseWallet) DecryptMessage(encMessage []byte, toVerKey string) ([]byte,
 
 // PackMessage Pack a message for one or more recipients.
 func (w *BaseWallet) PackMessage(envelope *Envelope) ([]byte, error) {
-	return nil, fmt.Errorf("not implemented")
+	if envelope == nil {
+		return nil, errors.New("envelope argument is nil")
+	}
+	// get keypair from db
+	key, err := w.getKey(envelope.FromVerKey)
```

Comment:
probably rename `key` to `senderKeyPair` or `fromKeyPair` as there are many keys involved in this function, for the sake of clarity

Reply:
done